config/v1/types_infrastructure.go

```diff
@@ -87,6 +92,22 @@ const (
 	VSpherePlatformType PlatformType = "VSphere"
 )
 
+// PlatformStatus holds the current status specific to the underlying infrastructure provider
+// of the current cluster. Since these are used at status-level for the underlying cluster, it
+// is supposed that only one of the status structs is set.
+type PlatformStatus struct {
+	// AWSPlatformStatus contains settings specific to the Amazon Web Services infrastructure provider.
+	// +optional
+	AWSPlatformStatus *AWSPlatformStatus `json:"awsPlatformStatus,omitempty"`
```

Comment:
/s/AWSPlatformStatus/AWS/ (and `aws` for the JSON tag)

do we need a pointer here?

Reply:
> /s/AWSPlatformStatus/AWS/ (and `aws` for the JSON tag)

Done with 532e4df7 -> e9a98202.

> do we need a pointer here?

Yeah, so we don't render `aws: {}` under `platformStatus` once we grow additional platforms here.  We need pointers for Go to respect `omitempty` for structures, golang/go#22480.

Reply:
I don't think this is the case - if we have `// +optional` in the go description, the apiserver doesn't render it out and you don't see it in `oc get` or `oc describe`.

See as an example `oc get image.config.openshift.io/cluster` - structs like `RegistrySources` don't render if they aren't set.

Reply:
Resolving this thread to consolidate pointer discussion under [this one](https://github.com/openshift/api/pull/300#discussion_r279091933).